test/paws_test.js

```diff
@@ -177,6 +202,67 @@ describe('Unit Tests', function() {
         setEnvStub.restore();
         responseStub.restore();
     });
+
+    describe('Load function', function() {
+        let fileWriteStub;
+        let fileReadStub;
+        let fileExistsStub;
+        const CREDS_FILE_PATH = '/tmp/paws_creds.json';
+        
+        beforeEach(() => {
+            fileWriteStub = sinon.spy(fs, 'writeFileSync');
+            fileReadStub = sinon.spy(fs, 'readFileSync');
+        });
+
+        afterEach(() => {
+            fileWriteStub.restore();
+            fileReadStub.restore();
+        });
+
+        it('gets creds from s3 if present and writes them to cache', function(done){
+            const oldAuthType = process.env.paws_auth_type;
+            process.env.paws_auth_type = 's3object';
+            fileExistsStub = sinon.stub(fs, 'existsSync').callsFake(() => {
+                return false;
+            });
+
+            TestCollector.load().then(function({pawsCreds}){
```

Comment:
I'd be great to verify the value of `pawsCreds` in every test.

Reply:
That is an issue that caused a lot of trouble when I was writing these tests...

The Paws collector class has some in-memory cacheing of the credentials, and my efforts to clear this cache after every test have been in vain. This cache means that the actual value of the creds cant be tested between tests. So we have to test whether the collecctor is at least going through the motions of using the file, even if its ultimatly going to use the value cached in memory at the end. 

This is apparently an issue with JS tests generally. I tries re-requireing the file and even tried using a library that claims to clear that cache. No dice.

So from what i can see, we would need to change the implementation fo that cache, or add method to clear it in order to test the creds value. Both of those options sounded like the tail wagging the dog, so I gave up trying.

Reply:
This is the cache for reference.

https://github.com/alertlogic/paws-collector/blob/master/paws_collector.js#L20

Reply:
let's try to use `rewire` tool to clean the in memory cache value.